src/rgw/rgw_rest_client.cc

```diff
@@ -61,34 +63,35 @@ int RGWHTTPSimpleRequest::receive_header(void *ptr, size_t len)
       *p = '\0';
       ldpp_dout(this, 30) << "received header:" << line << dendl;
       // TODO: fill whatever data required here
-      char *l = line;
+      char *l = line.data();
       char *tok = strsep(&l, " \t:");
       if (tok && l) {
         while (*l == ' ')
           l++;
- 
+
         if (strcmp(tok, "HTTP") == 0 || strncmp(tok, "HTTP/", 5) == 0) {
           http_status = atoi(l);
           if (http_status == 100) /* 100-continue response */
             continue;
           status = rgw_http_error_to_errno(http_status);
         } else {
           /* convert header field name to upper case  */
-          char *src = tok;
-          char buf[len + 1];
-          size_t i;
-          for (i = 0; i < len && *src; ++i, ++src) {
-            switch (*src) {
-              case '-':
-                buf[i] = '_';
-                break;
-              default:
-                buf[i] = toupper(*src);
-            }
-          }
-          buf[i] = '\0';
-          out_headers[buf] = l;
-          int r = handle_header(buf, l);
+	  std::string_view src(tok, len);
+	  std::string key;
+	  key.reserve(len);
+	  std::transform(src.cbegin(), src.cend(),
```

Comment:
Yeah, this pattern is definitely common enough that (I know you're probably tired of tinkering with this!) it would be good to see some kind of up/low-er_underscise() to have it in one place (strictly IMO though).